src/OpenTelemetry.Instrumentation.EntityFrameworkCore/EntityFrameworkInstrumentationOptions.cs

```diff
@@ -33,4 +35,15 @@ public class EntityFrameworkInstrumentationOptions
     /// Gets or sets a value indicating whether or not the <see cref="EntityFrameworkInstrumentation"/> should add the text of <see cref="CommandType.Text"/> commands as the <see cref="SemanticConventions.AttributeDbStatement"/> tag. Default value: False.
     /// </summary>
     public bool SetDbStatementForText { get; set; } = false;
+
+    /// <summary>
+    /// Gets or sets an action to enrich an Activity.
+    /// </summary>
+    /// <remarks>
+    /// <para><see cref="Activity"/>: the activity being enriched.</para>
+    /// <para>string: the name of the event.</para>
+    /// <para>object: the raw object from which additional information can be extracted to enrich the activity.
+    /// The type of this object depends on the event, which is given by the above parameter.</para>
+    /// </remarks>
+    public Action<Activity, string, object> Enrich { get; set; }
```

Comment:
I would change it to 
`public Action<Activity, IDbConnection?> EnrichWithIDbConnection { get; set; }`

and the line with calling to
`this.options.EnrichWithIDbConnection ?.Invoke(activity, connection as IDbConnection);`

I think this version will be easier to the end user.
What is more it is what we have for HttpInstrumentation. See https://github.com/search?q=repo%3Aopen-telemetry%2Fopentelemetry-dotnet+EnrichWith&type=code